### PR TITLE
Stream events by name

### PR DESCRIPTION
**Pull request by nlordell**
This PR implements streaming of events by name on generated contracts.

### Test Plan

CI with example

### CODE REVIEW

File: generate/src/contract/events.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        .map(|event| expand_filter(cx, event))
+        .collect::<Result<Vec<_>>>()?;
+    if filters.is_empty() {
+        return Ok(quote! {});
+    }
+
+    Ok(quote! {
+        impl #contract_name {
+            /// Retrieves a handle to a type containing for creating event
+            /// streams for all the contract events.
```
Comment (e00E) [inline generate/src/contract/events.rs]: This sentences reads weirdly. Maybe remove "containing"?

**Review by e00E — APPROVED** [review]

File: generate/src/contract/events.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    let ethcontract = &cx.runtime_crate;
+    let bytes = hash.as_bytes().iter().copied().map(Literal::u8_unsuffixed);
+
+    quote! {
+        #ethcontract::H256::from([#( #bytes ),*])
+    }
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (fleupold) [inline generate/src/contract/events.rs:108]: Might be useful to also add some tests for the remaining expansion logic?

**Review by fleupold — APPROVED** [review]
Looks really good. My only ask would be some more tests for the generated piece of code. It really helps future maintainers to understand what is being generated.

File: generate/src/contract/events.rs (hunk 2 above)
Comment (nlordell) [reply generate/src/contract/events.rs:108]: I can add a couple, the issue is sometimes tests just end up being verbose copies of the code and don't add much value (as it doesn't actually verify that the code is correct, just that it is being generated as expected). 

Specifically here, the `expand_filters` is of particular value.

Note that, these do get tested in the examples (we have an example that reads an event from the event stream) and the code gets generated for every contract example as well - so we test that the generated code actually compiles.

**Pull request merged**